lib/graphics.js

```diff
@@ -162,11 +162,17 @@ graphics.prototype.isPushed = function (page, bank, cb) {
 	cb(self.pushed[page + '_' + bank])
 }
 
-graphics.prototype.generate = function () {
+graphics.prototype.generate = function (invalidate) {
 	var self = this
 
 	for (var p = 1; p <= 99; p++) {
-		self.drawPage(p)
+			self.drawPage(p)
+			//Invalidate all on page if required
+			if (invalidate) {
+				for (var b = 1; b <= 32; b++) {
+					system.emit('graphics_bank_invalidated', p, b)
```

Comment:
Sorry, I meant to say to pass the invalidate parameter onto drawPage, and do the check below the `self.drawBank(page, bank)` line

Reply:
Ahh I see.
Will do